Readability.js

```diff
@@ -624,6 +624,11 @@ Readability.prototype = {
 
     var pageCacheHtml = page.innerHTML;
 
+    var articleTags = this._doc.getElementsByTagName("article");
+    if (articleTags.length === 1) {
+      this._doc.body.innerHTML = articleTags[0].parentNode.innerHTML;
```

Comment:
I went with

```js
var articleTags = this._doc.getElementsByTagName("article");
    if (articleTags.length === 1) {
      this._forEachNode(page.children, function(pageChild) {
        page.removeChild(pageChild);
      });
      page.appendChild(articleTags[0]);
    }
```

It seems to work well, though your solution is definitely shorter :)